code/modules/wiremod/components/bci/reagent_injector.dm

```diff
@@ -0,0 +1,54 @@
+/**
+ * # Reagent Injector Component
+ *
+ * Injects reagents into the user.
+ * Requires a BCI shell.
+ */
+
+/obj/item/circuit_component/reagent_injector
+	display_name = "Reagent Injector"
+	desc = "A component that can inject reagents from a BCI's reagent storage."
+	category = "BCI"
+	circuit_flags = CIRCUIT_NO_DUPLICATES
+
+	required_shells = list(/obj/item/organ/internal/cyberimp/bci)
+
+	var/datum/port/input/inject
+	var/datum/port/output/injected
+
+	var/obj/item/organ/internal/cyberimp/bci/bci
+
+/obj/item/circuit_component/reagent_injector/Initialize(mapload)
+	. = ..()
+	create_reagents(15, OPENCONTAINER) //This is mostly used in the case of a BCI still having reagents in it when the component is removed.
+
+/obj/item/circuit_component/reagent_injector/populate_ports()
+	. = ..()
+	inject = add_input_port("Inject", PORT_TYPE_SIGNAL, trigger = PROC_REF(trigger_inject))
+	injected = add_output_port("Injected", PORT_TYPE_SIGNAL)
+
+/obj/item/circuit_component/reagent_injector/proc/trigger_inject()
+	CIRCUIT_TRIGGER
+	if(!bci.owner)
+		return
+	if(bci.owner.reagents.total_volume + bci.reagents.total_volume > bci.owner.reagents.maximum_volume)
+		return
+	var/units = bci.reagents.trans_to(bci.owner.reagents, bci.reagents.total_volume, methods = INJECT)
```

Comment:
Add a `log_attack` (I think?? CC @dragomagol on what fits here best) to log that an injection happened, with who, and what reagents.

Reply:
Yup, `log_combat` is the typical reagent transfer logging proc

Reply:
Sure, will do once I'm not busy.